src/transports/ssh.c

```diff
@@ -893,9 +913,24 @@ int git_transport_ssh_with_paths(git_transport **out, git_remote *owner, void *p
 #endif
 }
 
+void git_transport_ssh_global_shutdown(void);
+
 int git_transport_ssh_global_init(void)
 {
 #ifdef GIT_SSH
+#ifdef GIT_WIN32
+	const char *(*libssh2_version)(int) = NULL;
+#else
+	libssh2_handle = dlopen("libssh2.dylib", RTLD_LAZY);
+	const char *(*libssh2_version)(int) = dlsym(libssh2_handle, "libssh2_version");
+#endif
+
+	// Check that libssh2's version is sufficient for us
+	if (!libssh2_version || !libssh2_version(LIBSSH2_VERSION_NUM)) {
+		return 0;
+	}
```

Comment:
This check should not be necessary. As mentioned in the other comment, we want to load a specific SOVERSION and otherwise we'll fail to load the library.